streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -123,83 +125,59 @@
     private Map<Integer, Set<String>> nodeGroups = null;
 
     // TODO: this is only temporary for 2.0 and should be removed
-    public final Map<StoreBuilder, String> storeToSourceChangelogTopic = new HashMap<>();
+    private final Map<StoreBuilder, String> storeToSourceChangelogTopic = new HashMap<>();
 
-    public interface StateStoreFactory {
-        Set<String> users();
-        boolean loggingEnabled();
-        StateStore build();
-        String name();
-        boolean isWindowStore();
-        Map<String, String> logConfig();
-        long retentionPeriod();
-    }
-
-    private static abstract class AbstractStateStoreFactory implements StateStoreFactory {
+    public static class StateStoreFactory {
```

Comment:
Sounds good.

Reply:
Getting rid of the abstract class sounds good, though i'd probably keep the interface. Generally i think it is better design to code against interfaces rather than concrete implementations.

Reply:
I agree in principle, but (like everything) I think it needs to be balanced. I've seen projects that have a requirement that every class also has an interface, and it gets unwieldy.

In this case, the interface seems unnecessary because this class is only used inside of this file (and in this file's tests).

Reply:
Also apologies for including this refactor in this PR. It was formerly related, but it's not anymore.

I'd pull it out, but I think all the reviewers have looked at it already.